tortoisebeacon/handlers.go

```diff
@@ -148,16 +147,51 @@ func (tb *TortoiseBeacon) handleProposalMessage(ctx context.Context, m ProposalM
 	return nil
 }
 
-func (tb *TortoiseBeacon) isPotentiallyValidProposalMessage(atxTimestamp time.Time, nextEpochStart time.Time, receivedTimestamp time.Time) bool {
+func (tb *TortoiseBeacon) verifyProposalMessage(m ProposalMessage, currentEpoch types.EpochID) (done bool, err error) {
+	currentEpochProposal, err := tb.buildProposal(currentEpoch)
+	if err != nil {
+		return true, fmt.Errorf("calculate proposal: %w", err)
+	}
+
+	ok := tb.vrfVerifier(m.MinerID.VRFPublicKey, currentEpochProposal, m.VRFSignature)
+	if !ok {
+		// TODO(nkryuchkov): attach telemetry
+		tb.Log.With().Warning("Received malformed proposal message: VRF is not verified",
+			log.String("sender", m.MinerID.Key))
+
+		// TODO(nkryuchkov): add a test for this case
+		return true, ErrMalformedProposal
+	}
+
+	epochWeight, _, err := tb.atxDB.GetEpochWeight(currentEpoch)
+	if err != nil {
+		return true, fmt.Errorf("get epoch weight: %w", err)
+	}
+
+	passes, err := tb.proposalPassesEligibilityThreshold(m.VRFSignature, epochWeight)
+	if err != nil {
+		return true, fmt.Errorf("proposalPassesEligibilityThreshold: %w", err)
+	}
+
+	if !passes {
+		tb.Log.With().Warning("Rejected proposal message which doesn't pass threshold")
+
+		return true, nil
```

Comment:
this case will cause us to print an error log message without actual message

Reply:
https://github.com/spacemeshos/go-spacemesh/pull/2553/files#diff-c301b2c1645d6588238bfe213b34a50d1ab81819fb56e68857529e9ec87eca3fR86
will return with error without inner error, is this expected? perhaps return it as a different error for consistency?

Reply:
No, it was a mistake. I don't like the signature of `verifyProposalMessage`, changed it